ARTICLE.md

```diff
@@ -1,27 +1,290 @@
+developers had provided `Route`'s. Both make managing external
+access to HTTP/HTTPS `Service`'s running on a cluster
+relatively easy. However, there are some notable differences,
+see the [Appendix](#ingress-vs-route) for a more in-depth
+comparison between `Route`'s and `Ingress`'s if you are
+interested.
+
+Overall, a `Route` in some ways is a slightly simpler
+abstraction which provides more robust support for handling
+TLS, and a greater separation of duties between the cluster
+administrators and developers. Additionally, with OpenShift
+3.10 there is now an ingress controller which will
+automatically make a `Route` (or possibly multiple) in the
+background for any `Ingress`'s you define.
+
+#### HTTPS Made Easy
+
+In pursuit of our goal of end-to-end TLS for our web services,
+we will take advantage of what `Route`'s have to offer.
+Firstly, using a `Route` we can easily expose an HTTP `Service`
+with a hostname under a certain path:
+
+```
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-component-v1
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+    version: v1
+spec:
+  type: ClusterIP
+  ports:
+    - port: 80
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: my-app
+    component: my-component
+    version: v1
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+```
+
+Even better, we can then secure our web service with edged
+terminated TLS and the ensure our users are always redirected
+to the HTTPS port of the load-balancer:
+
+```
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+so now our `Route` as a whole looks like:
+
+```
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+#### End-to-End TLS Made Easier
+
+Alright so that was pretty easy and we are a few steps closer
+to achieving end-to-end TLS. Moving forward, we first need our
+`Route` to no longer edge terminate TLS, which leaves us with
+two options for the termination policy: `passthrough` and
+`reencrypt`.
+
+With `passthrough`, the load-balancer will not participate in
+the TLS handshake and will simply send all traffic straight to
+the `Service` and then your `Pod`'s. That means your `Pod`'s
+would need a valid TLS cert for `my-component.apps.example.com`
+for the sake of the external client.
+
+With `reencrypt`, the load-balancer will handle the TLS
+handshake with the external client. Then it will do its own TLS
```

Comment:
So the load-balancer (I assume the `Route`?) needs it's own valid TLS cert?

Reply:
Yep and thats called out as an assumption in the beginning with some suggestions for getting that setup if thats not the case. I then at the end also mention how you can add the cert to the load-balancer/Route yourself if need be.

Reply:
Oh yep, you right